projects/hslayers/src/components/layermanager/layermanager.service.ts

```diff
@@ -925,13 +947,54 @@ export class HsLayerManagerService {
     }
   }
 
-  applyZIndex(layer): void {
+  /**
+   * Sets zIndex of layer being added.
+   * @param layer layer being added
+   */
+  private setPathMaxZIndex(layer: Layer): void {
+    //Sets zIndex to be highest in according path
+    let path = getPath(layer);
+    //If not set itll be assigned inside populateFolders function as 'other'
+    path = path ? path : 'other';
+
+    const pathLayers = this.data.layers.filter(
+      (layer) => getPath(layer.layer) == path
+    );
+
+    if (pathLayers.length > 0) {
+      //Get max avaialble index value
+      const maxPathZIndex = Math.max(
+        ...pathLayers.map((lyr) => lyr.layer.getZIndex() || 0)
+      );
+
+      layer.setZIndex(maxPathZIndex + 1);
+      //Increase zIndex of the layer that are supposed to be rendered above inserted
+      for (const lyr of this.data.layers.filter(
```

Comment:
Could have used HsLayermanagerPhysicalListService.moveTo

Reply:
Id also have to move the function out of the HsLayermanagerPhysicalListService becasue of the circular dependency

Reply:
right